aws/resource_aws_lambda_function.go

```diff
@@ -200,9 +200,19 @@ func resourceAwsLambdaFunction() *schema.Resource {
 
 			"tags": tagsSchema(),
 		},
+
+		CustomizeDiff: updateVerionIfPublish,
 	}
 }
 
+func updateVerionIfPublish(d *schema.ResourceDiff, meta interface{}) error {
+	publish := d.Get("publish").(bool)
+	if publish && needsFunctionCodeUpdate(d) {
```

Comment:
There are two other attributes that need to be SetNewComputed:

- last_modified - This need to be recomputed any time there is a change to the Lambda code, even if publish =false
- qualified_arn - This needs to be updated only if publish = true

Reply:
Thank you for the  feedback. I implemented the changes for last_modified and qualified_arn. In this version I only updated last_modified during the publish, but I see your point about it being needed more often than that.